tests/h/services/search_index/_queue_test.py

```diff
@@ -351,35 +336,96 @@ def search_index(
+        By default the annotation will exist in the DB but will *not* be in
+        Elasticsearch.
 
-    @pytest.fixture
-    def add_all(self, queue, annotation_ids):
-        def add_all(ids=annotation_ids, schedule_in=MINUS_5_MIN_IN_SECS, force=False):
-            for id_ in ids:
-                queue.add_by_id(
-                    id_, tag="test_tag", schedule_in=schedule_in, force=force
-                )
+        Usage:
+
+            SyncAnnotationJobFactory()
+
+            # Create multiple jobs, each with its own annotation.
+            SyncAnnotationJobFactory.create_batch(size=3)
+
+            # Create a job that isn't scheduled yet.
+            one_hour_from_now = datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+            SyncAnnotationJobFactory(scheduled_at=one_hour_from_now)
+
+            # Create an expired job.
+            one_hour_ago = datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+            SyncAnnotationJobFactory(expires_at=one_hour_ago)
+
+            # Create multiple jobs with force=True in their kwargs.
+            SyncAnnotationJobFactory.create_batch(size=3, force=True)
+
+            # Create a job for a given annotation, instead of
+            # SyncAnnotationJobFactory creating a new annotation automatically.
+            annotation = factories.Annotation()
+            SyncAnnotationJobFactory(annotation=annotation)
+
+            # Create a job for an annotation that *is* already in Elasticsearch.
+            SyncAnnotationJobFactory(index=True)
+        """
+
+        class SyncAnnotationJobFactory(factories.Job):
+            class Meta:
+                exclude = "force"
+
+            annotation = factory.SubFactory(factories.Annotation)
+            force = False
+
+            scheduled_at = factory.LazyFunction(
+                lambda: now - datetime_.timedelta(hours=1)
+            )
+            name = "sync_annotation"
+
+            kwargs = factory.LazyAttribute(
+                lambda o: {
+                    "annotation_id": URLSafeUUID.url_safe_to_hex(o.annotation.id),
+                    "force": o.force,
+                }
+            )
+
+            @classmethod
+            def _create(cls, model_class, *args, **kwargs):
+                annotation = kwargs.pop("annotation")
+                index_ = kwargs.pop("index", False)
+
+                job = super()._create(model_class, *args, **kwargs)
+
+                if index_:
+                    index(annotation)
+
+                return job
+
+        return SyncAnnotationJobFactory
 
-        return add_all
+    @pytest.fixture(autouse=True)
+    def noise_annotations(self, factories, index):
+        # Create some noise annotations in the DB. Some of them also in
+        # Elasticsearch, some not. None of these should ever be touched by the
+        # sync() method in these tests.
+        annotations = factories.Annotation.create_batch(size=2)
+        index(annotations[0])
```

Comment:
I realised that because there are no _other_ annotations in Elasticsearch it was possible to break the code and have the tests still pass.

Some noise jobs could be added to the DB for the same reason but doing so breaks a bunch of the tests above so I didn't bother.

Having other annotations in the DB actually doesn't cause a problem, even if the DB query is broken so that it doesn't filter them out.